### PR TITLE
update CT mount docs

### PR DESCRIPTION
**Pull request by marktnoonan**
Small updates to fix some examples to match new mount API, standardized wording around `MountReturn`.

### CODE REVIEW

**Review by elylucas — APPROVED** [review]

**Review by lmiller1990 — APPROVED** [review]

File: content/guides/component-testing/react/api.md [hunk 1]
```diff
@@ -123,7 +123,10 @@ it('mounts', () => {
 
 ### MountReturn
 
+Type that the `mount` function returns
```
Comment (emilyrohrbough) [inline content/guides/component-testing/react/api.md:126]: Shouldn't this be yielded, not returned?

Comment (lmiller1990) [reply content/guides/component-testing/react/api.md:126]: Ah yeah, good catch - wrong verbiage. 

Just to clarify for anyone reading/wondering about this

```js
cy.mount(<Foo />).then(val => ...)
```

The `val` is *yielded*. `cy.mount` _does_ return something, a `Chainable` (so you can do `cy.mount(...).get(...)`). The yielded value is what you get in the `then`.

Comment (marktnoonan) [reply content/guides/component-testing/react/api.md:126]: Yeah, I copied existing verbiage from the Angular, command, will update both.

Comment (marktnoonan) [reply content/guides/component-testing/react/api.md]: cool, updated everywhere

**Pull request merged**